src/data/datacube/seriesindex.cpp

```diff
@@ -0,0 +1,79 @@
+#include "seriesindex.h"
+
+#include "base/text/funcstring.h"
+
+using namespace Vizzu;
+using namespace Vizzu::Data;
+
+
+SeriesIndex::SeriesIndex(
+	const SeriesType &type,
+	const DataTable::DataIndex &dataIndex)
+	: index(dataIndex.value), type(type)
+{
+	if (type.isReal())
+	{
+		if (dataIndex.isInvalid())
+			throw std::logic_error
+			("series type needs valid column index");
+
+		if (	((dataIndex.type == ColumnInfo::Discrete) 
+				&& !type.isNestedDiscrete())
+			||	((dataIndex.type == ColumnInfo::Continous) 
+				&& type.isNestedDiscrete())
+		) throw std::logic_error
+				("invalid series type for discrete column");
+	}
+}
+
+SeriesIndex::SeriesIndex(const std::string &str, const DataTable &table)
+{
+	Text::FuncString func(str, false);
+
+	Data::SeriesType type;
+	if (!func.isEmpty())
+		type = Data::SeriesType::fromString(func.getName(), false);
+
+	if (type.isValid())
```

Comment:
If I recognize well, this equals to stg like this:
```cpp
if (Data::SeriesType type {func.isEmpty() ? Data::SeriesType() : Data::SeriesType::fromString(func.getName(), false)}; type.isValid())
```

I only afraid of variables scope leak (func and type).

This can be resolve without any action.

Reply:
done